link.go

```diff
@@ -1283,11 +1283,27 @@ func (ipoib *IPoIB) Type() string {
 	return "ipoib"
 }
 
+type Bareudp struct {
```

Comment:
Should _UDP_ be capitalized?
If I look at https://golang.org/pkg/net/ _TCP_ and _UDP_ are capitalized in many function/types names

Reply:
i have implemented the changes requested. you mind having a look again? 

thanks